lua/barbar/bbye.lua

```diff
@@ -237,13 +240,16 @@ function bbye.delete(action, force, buffer, mods)
   -- Using buflisted() over bufexists() because bufhidden=delete causes the
   -- buffer to still _exist_ even though it won't be :bdelete-able.
   if buflisted(buffer_number) == 1 and buffer_number ~= get_current_buf() then
-    local no_errors = pcall(cmd, action, buffer_number, force, mods)
-    if not no_errors then
-      if vim.v.errmsg:match('E516') then
-        set_current_buf(buffer_number)
+    local ok, msg = pcall(cmd, action, buffer_number, force, mods)
+    if not ok then
+      if msg then
+        if msg:match('E516') then
+          set_current_buf(buffer_number)
+        else
+          return err(msg)
+        end
       else
-        err(vim.v.errmsg)
-        return
+          return err('Could not delete buffer ' .. buffer_number .. ' with ' .. utils.markdown_inline_code(action))
```

Comment:
If `pcall` reports an error, but does not give a specific message, I have added this text.